In [1]:
# import the library
import pandas as pd

In [2]:
# load the Dataset and create train and test DataFrame  
df_train = pd.read_csv('gender-voice-train.csv')
df_test = pd.read_csv('gender-voice-test.csv')

In [3]:
# first five rows of training DataFrame
df_train.head(5)

label  meanfreq        sd    median       Q25       Q75       IQR  \
0      1  0.235668  0.035568  0.240065  0.222366  0.257763  0.035397   
1      0  0.190578  0.059717  0.190747  0.151985  0.242259  0.090273   
2      0  0.202016  0.062551  0.233775  0.133369  0.258403  0.125034   
3      1  0.236232  0.041141  0.244934  0.227932  0.260342  0.032410   
4      0  0.189003  0.061455  0.202376  0.135446  0.234851  0.099406   

       skew       kurt    sp.ent  ...      mode  centroid   meanfun    minfun  \
0  2.006333   6.427613  0.828406  ...  0.242788  0.235668  0.190731  0.047478   
1  2.727301  13.710214  0.901150  ...  0.186667  0.190578  0.118140  0.047198   
2  2.132116   7.588438  0.872074  ...  0.262950  0.202016  0.121069  0.046967   
3  2.175879   7.324086  0.831379  ...  0.243340  0.236232  0.179134  0.049896   
4  1.620451   6.374442  0.928735  ...  0.215050  0.189003  0.114454  0.047856   

     maxfun   meandom    mindom     maxdom    dfrange   modindx  
0  0.279070  1.240513  0.023438   9.140625   9.117188  0.087997  
1  0.279070  1.262109  0.023438   4.757812   4.734375  0.194908  
2  0.277457  0.898143  0.023438   8.460938   8.437500  0.085771  
3  0.274286  1.379012  0.023438  10.617188  10.593750  0.103917  
4  0.271186  1.147995  0.023438   5.953125   5.929688  0.119684  

[5 rows x 21 columns]

In [4]:
# first five rows of test DataFrame
df_test.head(5)

label  meanfreq        sd    median       Q25       Q75       IQR  \
0      1  0.186833  0.027472  0.184325  0.173955  0.204731  0.030777   
1      0  0.188879  0.060316  0.195537  0.138072  0.242975  0.104904   
2      0  0.150705  0.087127  0.174299  0.069666  0.226082  0.156416   
3      1  0.183667  0.040607  0.182534  0.156480  0.207646  0.051166   
4      1  0.205159  0.039543  0.210805  0.186667  0.228908  0.042241   

       skew       kurt    sp.ent  ...      mode  centroid   meanfun    minfun  \
0  2.655225  10.565846  0.821812  ...  0.180645  0.186833  0.172790  0.023495   
1  1.497393   5.037085  0.909425  ...  0.140386  0.188879  0.133092  0.050847   
2  2.603951  22.328899  0.969287  ...  0.050181  0.150705  0.109992  0.017260   
3  2.054138   7.483019  0.898138  ...  0.177040  0.183667  0.149237  0.018648   
4  2.099683   7.562209  0.876002  ...  0.224885  0.205159  0.154736  0.047105   

     maxfun   meandom    mindom     maxdom    dfrange   modindx  
0  0.271186  0.178571  0.007812   0.226562   0.218750  0.108929  
1  0.272727  0.855938  0.023438   8.718750   8.695312  0.098712  
2  0.266667  1.240954  0.007812   5.562500   5.554688  0.332396  
3  0.262295  0.550312  0.007812   3.421875   3.414062  0.166503  
4  0.277457  1.578835  0.187500  10.804688  10.617188  0.113644  

[5 rows x 21 columns]

In [5]:
# last five rows of training DataFrame
df_train.tail(5)

label  meanfreq        sd    median       Q25       Q75       IQR  \
2117      1  0.166738  0.052677  0.169427  0.162650  0.189223  0.026573   
2118      0  0.198718  0.058959  0.217333  0.143111  0.252000  0.108889   
2119      0  0.202333  0.063001  0.221946  0.137544  0.264817  0.127273   
2120      0  0.181799  0.058102  0.192037  0.123670  0.225568  0.101897   
2121      0  0.167732  0.066225  0.171886  0.112598  0.225196  0.112598   

          skew       kurt    sp.ent  ...      mode  centroid   meanfun  \
2117  7.550412  76.134526  0.865282  ...  0.167465  0.166738  0.152651   
2118  1.116666   3.569725  0.917123  ...  0.227556  0.198718  0.139322   
2119  2.000371   6.681799  0.873847  ...  0.272855  0.202333  0.123610   
2120  1.091660   4.009295  0.925575  ...  0.190731  0.181799  0.110586   
2121  0.822981   3.103282  0.953210  ...  0.173381  0.167732  0.126107   

        minfun    maxfun   meandom    mindom    maxdom   dfrange   modindx  
2117  0.022727  0.208333  0.174154  0.161133  0.214844  0.053711  0.136364  
2118  0.050473  0.279070  0.792092  0.023438  4.968750  4.945312  0.155766  
2119  0.047291  0.269663  1.190168  0.023438  7.429688  7.406250  0.093438  
2120  0.049741  0.274286  0.789062  0.023438  4.359375  4.335938  0.062312  
2121  0.048096  0.279070  0.813616  0.023438  6.023438  6.000000  0.123384  

[5 rows x 21 columns]

In [6]:
# last five rows of test DataFrame
df_test.tail(5)

label  meanfreq        sd    median       Q25       Q75       IQR  \
1041      0  0.143504  0.073334  0.150426  0.086383  0.197713  0.111330   
1042      0  0.151532  0.058071  0.151257  0.098246  0.195877  0.097632   
1043      1  0.230144  0.033577  0.231368  0.222526  0.246105  0.023579   
1044      0  0.204982  0.061316  0.234010  0.133237  0.254976  0.121739   
1045      0  0.171713  0.061671  0.189894  0.152660  0.207766  0.055106   

          skew       kurt    sp.ent  ...      mode  centroid   meanfun  \
1041  3.586009  28.346615  0.961517  ...  0.059947  0.143504  0.090472   
1042  3.523045  18.652713  0.893303  ...  0.094971  0.151532  0.095423   
1043  3.413691  17.897104  0.809968  ...  0.229158  0.230144  0.201065   
1044  1.850012   5.706426  0.867338  ...  0.249565  0.204982  0.120147   
1045  2.818282  11.990365  0.920936  ...  0.209255  0.171713  0.177866   

        minfun    maxfun   meandom    mindom    maxdom   dfrange   modindx  
1041  0.017448  0.235294  0.462015  0.007812  2.335938  2.328125  0.203020  
1042  0.056497  0.212766  0.519064  0.087891  0.776367  0.688477  0.650870  
1043  0.047059  0.279070  1.242622  0.234375  8.648438  8.414062  0.120313  
1044  0.018161  0.231884  0.408381  0.148438  0.796875  0.648438  0.312048  
1045  0.027165  0.271186  1.133854  0.031250  5.789062  5.757812  0.198301  

[5 rows x 21 columns]

In [7]:
# number of rows and columns in training DataFrame
row, col = df_train.shape
print('Row: ',row)
print('Col: ',col)

Row:  2122
Col:  21


In [8]:
# number of rows and columns in test DataFrame
row, col = df_test.shape
print('Row: ',row)
print('Col: ',col)

Row:  1046
Col:  21


In [9]:
# Check for the missing values in the training DataFrame
# False means there is no missing values

df_train.isnull().values.any()

False

In [10]:
# Check for the missing values in the test DataFrame
# False means there is no missing values

df_test.isnull().values.any()

False

In [11]:
# Count the number of male and female classes in the train DataFrame
# In the label column, 0 denotes the male class and 1 denotes the female class.

df_train['label'].value_counts()

1    1085
0    1037
Name: label, dtype: int64

In [12]:
# Separate the feature variables
X_train = df_train.iloc[:,1:].values 
X_test = df_test.iloc[:,1:].values 

In [13]:
# Separate the target variable
y_train = df_train.iloc[:,0].values
y_test  = df_test.iloc[:,0].values

In [14]:
# Fit the Random Forest Classifier to the Training set
from sklearn.ensemble import RandomForestClassifier
classifier = RandomForestClassifier(random_state = 42)
classifier.fit(X_train, y_train)

RandomForestClassifier(random_state=42)

In [15]:
# prediction on Test set
y_pred = classifier.predict(X_test)

In [16]:
# Confusion Matrix
from sklearn.metrics import confusion_matrix,classification_report,accuracy_score
cm = confusion_matrix(y_test, y_pred)
print(cm)

[[533  14]
 [  9 490]]


In [17]:
# classification report 
print(classification_report(y_test, y_pred))

              precision    recall  f1-score   support

           0       0.98      0.97      0.98       547
           1       0.97      0.98      0.98       499

    accuracy                           0.98      1046
   macro avg       0.98      0.98      0.98      1046
weighted avg       0.98      0.98      0.98      1046



In [18]:
# accuracy
acc = accuracy_score(y_test, y_pred)

In [19]:
# round() : round the values 10 3 decimal places 
print('accuracy: ',round(acc,3))

accuracy:  0.978


- Based on the confusion matrix, precision, recall ,f1-score and accuracy score , it is observed that the prediction model is making accurate prediction 97.8% of the time.